server/src/main/java/io/crate/expression/scalar/postgres/PgTableIsVisibleFunction.java

```diff
@@ -0,0 +1,111 @@
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.expression.scalar.postgres;
+
+import java.util.List;
+
+import io.crate.data.Input;
+import io.crate.exceptions.RelationUnknown;
+import io.crate.expression.scalar.HasTablePrivilegeFunction;
+import io.crate.metadata.FunctionName;
+import io.crate.metadata.Functions;
+import io.crate.metadata.NodeContext;
+import io.crate.metadata.RelationName;
+import io.crate.metadata.Scalar;
+import io.crate.metadata.Schemas;
+import io.crate.metadata.SearchPath;
+import io.crate.metadata.TransactionContext;
+import io.crate.metadata.functions.BoundSignature;
+import io.crate.metadata.functions.Signature;
+import io.crate.metadata.pgcatalog.PgCatalogSchemaInfo;
+import io.crate.metadata.settings.SessionSettings;
+import io.crate.metadata.table.Operation;
+import io.crate.role.Permission;
+import io.crate.role.Role;
+import io.crate.role.Roles;
+import io.crate.sql.tree.QualifiedName;
+import io.crate.types.DataTypes;
+
+public class PgTableIsVisibleFunction extends Scalar<Boolean, Integer> {
+
+    private static final FunctionName NAME = new FunctionName(PgCatalogSchemaInfo.NAME, "pg_table_is_visible");
+
+    public static void register(Functions.Builder module) {
+        module.add(
+            Signature.scalar(
+                NAME,
+                DataTypes.INTEGER.getTypeSignature(),
+                DataTypes.BOOLEAN.getTypeSignature()
+            ).withFeature(Feature.NULLABLE),
+            PgTableIsVisibleFunction::new);
+    }
+
+    protected PgTableIsVisibleFunction(Signature signature, BoundSignature boundSignature) {
+        super(signature, boundSignature);
+    }
+
+    @SafeVarargs
+    @Override
+    public final Boolean evaluate(TransactionContext txnCtx, NodeContext nodeContext, Input<Integer>... args) {
+        assert args.length == 1 : NAME + " expects exactly 1 argument, got " + args.length;
+        Integer tableOid = args[0].value();
+        if (tableOid == null) {
+            return null;
+        }
+
+        Schemas schemas = nodeContext.schemas();
+        RelationName relationName = schemas.getRelation(tableOid);
+        if (relationName == null) {
+            return false;
+        }
+
+        SessionSettings sessionSettings = txnCtx.sessionSettings();
+        Roles roles = nodeContext.roles();
+        Role user = roles.findUser(sessionSettings.userName());
+        for (String searchPathIdx : sessionSettings.searchPath()) {
+            try {
+                schemas.findRelation(
+                    QualifiedName.of(searchPathIdx, relationName.name()),
+                    Operation.READ,
+                    user,
+                    SearchPath.createSearchPathFrom(searchPathIdx));
+            } catch (RelationUnknown e) {
+                continue;
+            }
```

Comment:
Depending on try-catch mechanism is not suggested but thought it is too much to come up with a new method in `schemas` just for a scalar function.